sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/PropertyConfigurator.java

```diff
@@ -151,4 +159,39 @@ private String getAppLocale() {
     public void setAppLocale(String appLocale) {
         mAppLocale = appLocale;
     }
+
+    /**
+     * Add or overwrite the given key for the common event properties. Properties will be inherited
+     * by children of this transmission target.
+     *
+     * @param key   The property key.
+     * @param value The property value.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public synchronized void setEventProperty(String key, String value) {
+        mEventProperties.put(key, value);
+    }
+
+    /**
+     * Removes the given key from the common event properties.
+     *
+     * @param key The property key to be removed.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public synchronized void removeEventProperty(String key) {
+        mEventProperties.remove(key);
+    }
+
+    /**
+     * Extracted method to synchronize on each level at once while reading properties.
+     * Nesting synchronize between parent/child could lead to deadlocks.
+     */
+    protected synchronized void mergeEventProperties(Map<String, String> mergedProperties) {
```

Comment:
can you explain why private does not work? Who uses the method besides internally?

Reply:
It's used in `trackEvent` from `AnalyticsTransmissionTarget` when we're iterating through parents and merging event properties: https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/AnalyticsTransmissionTarget.java#L91

Reply:
You can remove protected then and use no keyword